### ISSUE TITLE
[ONNX] Be able to fold constant Softmax nodes during export

### ISSUE THREAD

**Issue by ilyasher** [pytorch/pytorch#97927]
### 🚀 The feature, motivation and pitch

Please consider my toy model, which has a softmax which is independent of any input:

```python
import torch

class MyModel(torch.nn.Module):

    def __init__(self):
        super().__init__()
        self.vec = torch.nn.Parameter(torch.randn(10, 1000))


    def forward(self, x):
        # Does not depend on input, Can be precomputed.
        norm = torch.softmax(self.vec, -1)

        # Depends on input, cannot be precomputed.
        return x + norm


model = MyModel()
with torch.no_grad():
    torch.onnx.export(model, (torch.randn(10, 1000),), 'const-fold.onnx',
        input_names=['input'],
        output_names=['output'],
        do_constant_folding=True,
    )
```

However, even with `do_constant_folding=True`, the softmax node does not get folded:

![no-fold](https://user-images.githubusercontent.com/46343317/228689128-2f39a839-fe6e-4801-a7f2-651aa1cb0ff5.png)

Softmax preserves tensor volume so folding it shouldn't bloat the produced ONNX file.

It seems like this would be a matter of adding Softmax to [this giant if/else block](https://github.com/pytorch/pytorch/blob/7e43c6f28e9b0699dd6f0a3803513feacc60eaf3/torch/csrc/jit/passes/onnx/constant_fold.cpp#L262-L535).

Maybe there are also some other ops which constant folding could be supported for. Matmul would be nice but it has the potential to produce huge tensors in the exported ONNX model.

Thanks!
